### PR TITLE
feat: `updateGroupingPolicy()`, `updateNamedGroupingPolicy()`, `getImplicitResourcesForUser()`

### PR DESCRIPTION
**Pull request by Shivansh-yadav13**
- ref #324

### CODE REVIEW

**Comment by hsluoyz** [comment]
@nodece

File: src/enforcer.ts [hunk 1]
```diff
@@ -420,6 +420,44 @@ export class Enforcer extends ManagementEnforcer {
...
+    const permissions = await this.getImplicitPermissionsForUser(user, ...domain);
+    const res: string[][] = [];
+    for (const permission of permissions) {
+      if (permission[0] === user) {
+        res.push(permission);
+        continue;
+      }
+      let resLocal: string[][] = [[user]];
+      const tokensLength: number = permission.length;
+      const t: string[][] = [[]];
```
Comment (nodece) [inline src/enforcer.ts:437]:
```suggestion
      const t: string[][] = [];
```

**Comment by hsluoyz** [comment]
@Shivansh-yadav13 plz fix:

image

**Comment by hsluoyz** [comment]
@Shivansh-yadav13 btw, plz join here:

image

**Comment by Shivansh-yadav13** [comment]
@hsluoyz fixed CI

**Comment by hsluoyz** [comment]
@nodece @Zxilly plz review

**Comment by hsluoyz** [comment]
Replaced by: https://github.com/casbin/node-casbin/pull/419

**Pull request closed without merging**